curriculum/challenges/english/20-upcoming-python/learn-list-comprehension-by-building-a-case-converter-program/657f43d341a0dd17120cdb08.md

```diff
@@ -0,0 +1,68 @@
+---
+id: 657f43d341a0dd17120cdb08
+title: Step 17
+challengeType: 20
+dashedName: step-17
+---
+
+# --description--
+
+Inside the space you left between the pair of square braces, you can describe the value that you would like to include in the list based on a given condition.
+
+```py
+snake_cased_char_list = [
+    '_' + char.lower() if char.isupper()
+]
+```
+
+Python will interpret this expression as "append `'_' + char.lower()` to the list if `char` is in uppercase" and this will convert the case for the capital letters in the input string.
+
+Start by adding this line within the square braces.
+
+# --hints--
+
+You should add `'_' + char.lower() if char.isupper()` within the square braces of `snake_cased_char_list`.
+
+```js
+({
+    test: () => {
+        const transformedCode = e.code.original["main.py"].replace(/\r/g, "");
+        const convert_to_snake_case = __helpers.python.getDef("\n" + transformedCode, "convert_to_snake_case");
+        const { function_body } = convert_to_snake_case;
+
+        assert.match(function_body, / +snake_cased_char_list\s*=\s*\[\n*\s*'_'\s*\+\s*char.lower\(\)\s{1}if\s{1}char.isupper\(\)\n*\]/);
+    }
+})
```

Comment:
```suggestion
const transformedCode = e.code.original["main.py"].replace(/\r/g, "");
const convert_to_snake_case = __helpers.python.getDef("\n" + transformedCode, "convert_to_snake_case");
const { function_body } = convert_to_snake_case;

assert.match(function_body, / +snake_cased_char_list\s*=\s*\[\n*\s*'_'\s*\+\s*char.lower\(\)\s+if\s+char.isupper\(\s*\)\n*\]/);
```

Reply:
Since the code is incomplete it throws an error and that test format is not suitable.